src/main/java/net/earthcomputer/clientcommands/command/AutoPrefixCommand.java

```diff
@@ -0,0 +1,60 @@
+package net.earthcomputer.clientcommands.command;
+
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.minecraft.commands.CommandBuildContext;
+import net.minecraft.network.chat.Component;
+
+import static dev.xpple.clientarguments.arguments.CComponentArgument.getComponent;
+import static dev.xpple.clientarguments.arguments.CComponentArgument.textComponent;
+import static dev.xpple.clientarguments.arguments.CGameProfileArgument.gameProfile;
+import static dev.xpple.clientarguments.arguments.CGameProfileArgument.getSingleProfileArgument;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.argument;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.literal;
+
+public class AutoPrefixCommand {
+    private static String currentPrefix = null;
+
+    public static String getCurrentPrefix() {
+        return currentPrefix;
+    }
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher, CommandBuildContext context) {
+        dispatcher.register(literal("cautoprefix")
+                .then(literal("set")
+                        .then(literal("tmsg")
+                                .executes(ctx -> prefix(ctx.getSource(), "/tm")))
+                        .then(literal("msg")
+                                .then(argument("player", gameProfile(true))
+                                    .executes(ctx -> prefix(ctx.getSource(), "/w " + getSingleProfileArgument(ctx, "player").getName()))))
+                        .then(argument("prefix", textComponent(context))
```

Comment:
As discussed on discord, `greedyString` here. It should definitely not be a text component, as that's not what you're typing in the text box.